src/poliastro/czml/extract_czml.py

```diff
@@ -100,6 +100,52 @@ def _init_orbit_packet_cords_(self, i, rtol):
 
         return cart_cords
 
+    def _init_groundtrack_packet_cords_(self, i, rtol):
+        """
+
+        Parameters
+        ----------
+        i: int
+            Index of referenced orbit
+        rtol: float
+            Maximum relative error permitted
+
+        Returns
+        -------
+        coordinate list
+        """
+        cart_cords = []  # type: List[float]
+
+        h = (self.end_epoch - self.orbits[i][2]).to(u.second) / self.orbits[i][1]
+
+        # Get rounding factor given the relative tolerance
+        rf = 0
+        while rtol < 1:
+            rtol *= 10
+            rf += 1
+
+        ellipsoid = self.cust_prop[0]
+
+        for k in range(self.orbits[i][1] + 2):
+            position = propagate(self.orbits[i][0], TimeDelta(k * h), rtol=rtol)
+
+            cords = position.represent_as(CartesianRepresentation).xyz.to(u.meter).value
+            cords = np.insert(cords, 0, h.value * k, axis=0)
+
+            # flatten list
+            cords = list(map(lambda x: round(x[0], rf), cords.tolist()))
+            t, p = cords[0], cords[1:]
+            pr_p = project_point_on_ellipsoid(
+                p[0], p[1], p[2], ellipsoid[0], ellipsoid[1], ellipsoid[2]
+            )
+            #  This is a hack to ensure that our point lies above the surface of the
+            # ellipsoid. An iterative method could be used instead but the error margin
+            # is too small to be worth it.
```

Comment:
Good to have a docstring here - could you comment a bit on how the hack works though? Not sure what `+ 0.1` does (to clarify: could you expand the code comment?)

Reply:
Ok, I'll try to make it more clear. The added constant simply ensures that the point lies above the surface. The reason we have to do this, is because the resulting point may be slightly bellow the surface because of numerical errors (it should be rounded up anyway, but better be sure that no graphical glitch will result out of this)